docs/app/components/SampleGallery/SampleGallery.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+
+import useStyles from './SampleGallery.styles';
+import SampleCard from '../SampleCard/SampleCard';
+import { FC, useEffect, useState } from 'react';
+import { parse } from 'yaml';
+import config from '../../../next.config';
+
+export interface Sample {
+  id: string;
+  title: string;
+  description: string;
+  tags: string[];
+  githubUrl: string;
+  imageUrl: string;
+  author: string;
+  language: string;
+  demoUrlGif: string;
+  longDescription: string;
+  featuresList: string[];
+}
+
+interface SamplesData {
+  samples: Sample[];
+}
+
+const resolveImageUrl = (imageUrl: string) => {
+  // If the image URL is relative, prepend the base path
+  if (imageUrl.startsWith('/')) {
+    return `${config.basePath}${imageUrl}`;
+  }
+  return imageUrl;
+};
+
+const SampleGallery: FC = () => {
+  const classes = useStyles();
+  const [samples, setSamples] = useState<Sample[]>([]);
+
+  useEffect(() => {
+    async function loadSamples() {
+      try {
+        const response = await fetch(`${config.basePath}/data/samples.yaml`);
```

Comment:
Can we compute the renderable remark before? basically when we build samples?

Reply:
it is rendered in build time now